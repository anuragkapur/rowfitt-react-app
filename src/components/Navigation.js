import React from 'react';
import {
  BrowserRouter as Router,
  Route,
  Link
} from 'react-router-dom';
import Dashboard from './Dashboard';
import LogNewWorkout from './LogNewWorkout'
import ViewTrainingLog from './ViewTrainingLog'
import LogSingleDistanceWorkout from "./LogSingleDistanceWorkout";
import UploadWorkoutImage from "./UploadWorkoutImage";
import Auth from '../auth/Auth';

const auth = new Auth();

class Navigation extends React.Component {

  login() {
    auth.login();
  }

  logout() {
    auth.logout();
  }

  render() {
    const { isAuthenticated } = auth;

    return (
      <Router>
        <div>
          <header>
            <nav className="navbar navbar-expand-md navbar-dark fixed-top bg-dark">
              <a className="navbar-brand" href="/">Rowfitt</a>
              <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarCollapse"
                      aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation">
                <span className="navbar-toggler-icon"></span>
              </button>
              <div className="collapse navbar-collapse" id="navbarCollapse">
                <ul className="navbar-nav mr-auto">
                  <li className="nav-item">
                    <Link className="nav-link" to="/">Dashboard</Link>
                  </li>
                  <li className="nav-item">
                    <Link className="nav-link" to="/log-new-workout">Log New Workout</Link>
                  </li>
                  <li className="nav-item">
                    <Link className="nav-link" to="/view-training-log">View Training Log</Link>
                  </li>
                  <li className="nav-item-active">
                    {
                      !isAuthenticated() && (<Link className="nav-link" to="#" onClick={this.login.bind(this)}>Login</Link>)
                    }
                    {
                      isAuthenticated() && (<Link className="nav-link" to="#" onClick={this.logout.bind(this)}>Logout</Link>)
                    }
                  </li>
                </ul>
              </div>
            </nav>
          </header>
          <Route exact path="/" component={Dashboard}/>
          <Route exact path="/log-new-workout" component={LogNewWorkout}/>
          <Route exact path="/view-training-log" component={ViewTrainingLog}/>
          <Route exact path="/log-new-workout/single-distance-workout" component={LogSingleDistanceWorkout}/>
          <Route exact path="/log-new-workout/upload-image" component={UploadWorkoutImage}/>
        </div>
      </Router>
    );
  }
}

export default Navigation;