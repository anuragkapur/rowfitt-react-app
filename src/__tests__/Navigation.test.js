import React from 'react';
import Navigation from "../components/Navigation";
import {render, fireEvent, cleanup} from 'react-testing-library';
import 'jest-dom/extend-expect';

afterEach(cleanup);

test("navigation elements - Dashboard, LogNewWorkout, ViewTrainingLogs are displayed", () => {
  const {getByText, container} = render(
    <Navigation/>
  );

  expect(getByText("Dashboard")).toBeInTheDocument();
  expect(getByText("Log New Workout")).toBeInTheDocument();
  expect(getByText("View Training Log")).toBeInTheDocument();
  expect(container.firstChild).toMatchSnapshot();
});

test("clicking on / link displays Dashboard page", () => {
  const {getByText, container} = render(
    <Navigation/>
  );
  const leftClick = {button: 0};
  fireEvent.click(getByText("Dashboard"), leftClick);
  expect(container.innerHTML).toMatch("This is the dashboard");
});

test("clicking on /log-new-workout link displays LogNewWorkout page", () => {
  const {getByText, container} = render(
    <Navigation/>
  );
  const leftClick = {button: 0};
  fireEvent.click(getByText("Log New Workout"), leftClick);
  expect(container.innerHTML).toMatch("Log a New Workout");
});

test("clicking on /view-training-log link displays ViewTrainingLog page", () => {
  const {getByText, container} = render(
    <Navigation/>
  );
  const leftClick = {button: 0};
  fireEvent.click(getByText("View Training Log"), leftClick);
  expect(container.innerHTML).toMatch("My Training Log");
});

test("navigation from top nav to 'Log New Workout' link to 'Single Distance' link takes user to " +
  "LogSingleDistanceWorkout page", () => {
  const {getByText, container} = render(<Navigation/>);
  const leftClick = {button: 0};
  fireEvent.click(getByText("Log New Workout"), leftClick);
  fireEvent.click(getByText("Single Distance"), leftClick);
  expect(getByText("Log Single Distance Workout")).toBeInTheDocument();
  expect(container.firstChild).toMatchSnapshot();
});