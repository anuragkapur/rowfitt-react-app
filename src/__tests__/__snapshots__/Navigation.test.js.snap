// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navigation elements - Dashboard, LogNewWorkout, ViewTrainingLogs are displayed 1`] = `
<div>
  <header>
    <nav
      class="navbar navbar-expand-md navbar-dark fixed-top bg-dark"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        Rowfitt
      </a>
      <button
        class="navbar-toggler"
        type="button"
        data-toggle="collapse"
        data-target="#navbarCollapse"
        aria-controls="navbarCollapse"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
        id="navbarCollapse"
      >
        <ul
          class="navbar-nav mr-auto"
        >
          <li
            class="nav-item active"
          >
            <a
              class="nav-link"
              href="/"
            >
              Dashboard 
              <span
                class="sr-only"
              >
                (current)
              </span>
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/log-new-workout"
            >
              Log New Workout
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/view-training-log"
            >
              View Training Log
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </header>
  <div>
    <br />
    <br />
    <br />
    <h2>
      This is the dashboard
    </h2>
  </div>
</div>
`;

exports[`navigation from top nav to 'Log New Workout' link to 'Single Distance' link takes user to LogSingleDistanceWorkout page 1`] = `
<div>
  <header>
    <nav
      class="navbar navbar-expand-md navbar-dark fixed-top bg-dark"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        Rowfitt
      </a>
      <button
        class="navbar-toggler"
        type="button"
        data-toggle="collapse"
        data-target="#navbarCollapse"
        aria-controls="navbarCollapse"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
        id="navbarCollapse"
      >
        <ul
          class="navbar-nav mr-auto"
        >
          <li
            class="nav-item active"
          >
            <a
              class="nav-link"
              href="/"
            >
              Dashboard 
              <span
                class="sr-only"
              >
                (current)
              </span>
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/log-new-workout"
            >
              Log New Workout
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/view-training-log"
            >
              View Training Log
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </header>
  <div>
    <br />
    <br />
    <br />
    <br />
    <h2>
      Log Single Distance Workout
    </h2>
  </div>
</div>
`;
